SnowplowSwiftDemo/SnowplowSwiftDemo/DemoUtils.swift

```diff
@@ -101,36 +101,36 @@ class DemoUtils {
         tracker.trackUnstructuredEvent(event)
     }
     
-    /**static func trackScreenViewWithTracker(_ tracker: SPTracker) {
+    static func trackScreenViewWithTracker(_ tracker: SPTracker) {
         var event = SPScreenView.build({ (builder : SPScreenViewBuilder?) -> Void in
             builder!.setName("DemoScreenName")
-            builder!.setId("DemoScreenId")
+            builder!.setEventId("DemoScreenId")
```

Comment:
I think these should be setScreenId as you are constructing a Screen View event and they need an ID as per the screen_view schema: https://github.com/snowplow/iglu-central/blob/master/schemas/com.snowplowanalytics.mobile/screen_view/jsonschema/1-0-0

Also, if you look in SPEvent's constructor - it sets a event id so it should already have a valid UUID set.